src/lib/parser.rs

```diff
@@ -100,9 +100,18 @@ impl Parser {
     Err(res)
   }
   pub fn parse(contents: impl Into<Vec<u8>>) -> Result<Self, ParsingError> {
+    // this removes \r as it seems to cause problems during parsing
+    let mut tokens = contents.into();
+    for i in 0..tokens.len() {
+      let c = tokens.get(i);
+      match c {
+          Some(&13) => {tokens.remove(i);},
+          _ => ()
+      }
```

Comment:
I think this is a bit of a cleaner solution
```suggestion
      if let Some(&13) = tokens.get(i) {
          contents.remove(i);
      }
```

Reply:
Ah yes. I did change it, but then had a merging issue